src/test/java/picard/sam/AddOATagTest.java

```diff
@@ -0,0 +1,142 @@
+        firstPassOutput.deleteOnExit();
+        final File secondPassOutput = File.createTempFile("SecondPassAddOATag", ".bam");
+        secondPassOutput.deleteOnExit();
+
+        // make first pass bam that only has one value per OA tag
+        runAddOATag(inputSam, firstPassOutput, interval_list, true);
+
+        // make bam we want to test the overwrite option with
+        runAddOATag(firstPassOutput, secondPassOutput, interval_list, overWriteTag);
+
+        validateBamHasOA(secondPassOutput, interval_list, overWriteTag);
+    }
+
+    private void runAddOATag(final File inputSam, final File output, final File intervalList, Boolean overwriteTag) throws IOException {
+        final ArrayList<String> args = new ArrayList<String>(){
+            {
+                add("INPUT=" + inputSam);
+                add("OUTPUT=" + output);
+            }
+        };
+        if (intervalList != null) {
+            args.add("INTERVAL_LIST=" + intervalList);
+        }
+        if (overwriteTag != null) {
+            args.add("OVERWRITE_TAG=" + overwriteTag);
+        }
+        AddOATag addOATag = new AddOATag();
+        Assert.assertEquals(addOATag.instanceMain(args.toArray(new String[args.size()])), 0, "Running addOATag did not succeed");
+    }
+
+    private void validateOATAG(final File testSam, final File truthSam) {
+        final ArrayList<String> truthOAValues = new ArrayList<>();
+        final ArrayList<String> testOAValues = new ArrayList<>();
+
+        SAMRecordIterator iterator = SamReaderFactory.makeDefault().open(truthSam).iterator();
+        while (iterator.hasNext()){
+            SAMRecord rec = iterator.next();
+            truthOAValues.add(rec.getStringAttribute("OA"));
+        }
+
+        iterator = SamReaderFactory.makeDefault().open(testSam).iterator();
+        while (iterator.hasNext()){
+            SAMRecord rec = iterator.next();
+            testOAValues.add(rec.getStringAttribute("OA"));
+        }
+
+        Assert.assertTrue(truthOAValues.equals(testOAValues));
+    }
+
+    private void validateBamHasOA(final File input, final File intervalList, Boolean overwriteTag) {
+        final SAMRecordIterator iterator = SamReaderFactory.makeDefault().open(input).iterator();
+
+        OverlapDetector<Interval> detector = AddOATag.getOverlapDetectorFromIntervalListFile(intervalList, 0, 0);
+        while (iterator.hasNext()){
+            SAMRecord rec = iterator.next();
+            if (!rec.getReadUnmappedFlag() && (detector == null || detector.overlapsAny(rec))){
+                Assert.assertNotNull(rec.getAttribute("OA"));
+                if (overwriteTag != null){
+                    validateOverwriteTag(rec, overwriteTag);
+                }
+            } else {
+                Assert.assertNull(rec.getAttribute("OA"));
+            }
+        }
+    }
+
+    private void validateOverwriteTag(SAMRecord rec, Boolean overwriteTag) {
```

Comment:
I would suggest to have instead of a boolean an integer indicating how many OA tags are expected - this is more flexible for adding new tests.